plugins/logitech-bulkcontroller/proto_manager.c

```diff
@@ -0,0 +1,192 @@
+	  case LOGI__DEVICE__PROTO__RESPONSE__PAYLOAD_GET_DEVICE_INFO_RESPONSE:
+		  g_printf("[proto_manager_decode_message] - LOGI__DEVICE__PROTO__RESPONSE__PAYLOAD_GET_DEVICE_INFO_RESPONSE\n");
+		 if (usb_msg->response->get_device_info_response) {
+			 proto_id = kProtoId_GetDeviceInfoResponse;
+			 if (usb_msg->response->get_device_info_response->payload) {
+				 g_stpcpy((gchar*)&decoded_data->device_info, usb_msg->response->get_device_info_response->payload);
+			 }
+		 }
+		  break;
+	  case LOGI__DEVICE__PROTO__RESPONSE__PAYLOAD_TRANSITION_TO_DEVICEMODE_RESPONSE:
+		  g_printf("[proto_manager_decode_message] - LOGI__DEVICE__PROTO__RESPONSE__PAYLOAD_TRANSITION_TO_DEVICEMODE_RESPONSE\n");
+			 if (usb_msg->response->transition_to_devicemode_response) {
+				 proto_id = kProtoId_TransitionToDeviceModeResponse;
+				 decoded_data->transition_to_device_mode = usb_msg->response->transition_to_devicemode_response->success;
+			 }
+		  break;
+	  default:
+		  g_printf("[proto_manager_decode_message] - Unhandled response %d\n", usb_msg->response->payload_case);
+		  break;
+	  };
+  }
+	  break;
+  case LOGI__DEVICE__PROTO__USB_MSG__MESSAGE_EVENT:
+  {
+	  if (!usb_msg->event)
+		  return proto_id;
+	  g_printf("[proto_manager_decode_message] - LOGI__DEVICE__PROTO__USB_MSG__MESSAGE_EVENT\n");
+	  switch (usb_msg->event->payload_case) {
+	  case LOGI__DEVICE__PROTO__EVENT__PAYLOAD_KONG_EVENT:
+	  {
+		  g_printf("[proto_manager_decode_message] - LOGI__DEVICE__PROTO__EVENT__PAYLOAD_KONG_EVENT\n");
+		 if (usb_msg->event->kong_event) {
+			 proto_id = kProtoId_KongEvent;
+			 if (usb_msg->event->kong_event->mqtt_event) {
+				 g_stpcpy((gchar*)&decoded_data->device_info, usb_msg->event->kong_event->mqtt_event);
+			 }
+		 }
+
+	  }
+		  break;
+	  case LOGI__DEVICE__PROTO__EVENT__PAYLOAD_HANDSHAKE_EVENT:
+	  {
+		  g_printf("[proto_manager_decode_message] - LOGI__DEVICE__PROTO__EVENT__PAYLOAD_HANDSHAKE_EVENT\n");
+		 if (usb_msg->event->handshake_event) {
+			 proto_id = kProtoId_HandshakeEvent;
+			 decoded_data->hand_shake_event = TRUE;
+		 }
+	  }
+		  break;
+	  case LOGI__DEVICE__PROTO__EVENT__PAYLOAD_CRASH_DUMP_AVAILABLE_EVENT:
+	  {
+		  g_printf("[proto_manager_decode_message] - LOGI__DEVICE__PROTO__EVENT__PAYLOAD_CRASH_DUMP_AVAILABLE_EVENT\n");
+		  proto_id = kProtoId_CrashDumpAvailableEvent;
+	  }
+		  break;
+	  default:
+		  g_printf("[proto_manager_decode_message] - Unhandled event %d\n", usb_msg->event->payload_case);
+		  break;
+	  };
+  }
+	  break;
+  default:
+	  g_printf("[proto_manager_decode_message] - Unknown\n");
+	  break;
+  };
+  logi__device__proto__usb_msg__free_unpacked(usb_msg,NULL);
+  return proto_id;
+}
+
+/*!
+* @brief Parse the device info
+* @param data - string containing the device info
+* @param dev_info - contains the device information
+* @return true if parsing is successful
+*/
+gboolean proto_manager_parse_device_info(const gchar* data, device_info* dev_info) {
+return TRUE;
+}
```

Comment:
the formatting is all off in this file -- use ./contrib/reformat-code.py

Reply:
Fixed